plugins/utils/alpha.nix

```diff
@@ -0,0 +1,232 @@
+        };
+
+        opts = mkOption {
+          type = types.submodule {
+            options = {
+              spacing = mkOption {
+                type = types.int;
+                default = 0;
+                description = "Spacing between grouped components";
+              };
+
+              hl = mkOption {
+                type = types.str;
+                default = "Keyword";
+                description = "HighlightGroup to apply";
+              };
+
+              position = mkOption {
+                type = types.str;
+                default = "center";
+                description = "How to align section";
+              };
+
+              margin = mkOption {
+                type = types.int;
+                default = 0;
+                description = "Margin for section";
+              };
+            };
+          };
+          default = {};
+          description = "Additional options for the section";
+        };
+      };
+    };
+in {
+  options = {
+    plugins.alpha = {
+      enable = mkEnableOption "alpha";
+
+      package = helpers.mkPackageOption "alpha" pkgs.vimPlugins.alpha-nvim;
+
+      theme = mkOption {
+        description = "Optional theme for alpha-nvim, one of: dashboard, startify, theta";
+        type = themeType;
+        default = "dashboard";
+      };
+
+      iconsEnabled = mkOption {
+        type = types.bool;
+        description = "Toggle icon support. Installs nvim-web-devicons.";
+        default = true;
+      };
+
+      layout = with helpers;
+        mkOption {
+          default = [
+            {
+              type = "padding";
+              val = 2;
+            }
+            {
+              type = "text";
+              val = [
+                "  ███╗   ██╗██╗██╗  ██╗██╗   ██╗██╗███╗   ███╗  "
+                "  ████╗  ██║██║╚██╗██╔╝██║   ██║██║████╗ ████║  "
+                "  ██╔██╗ ██║██║ ╚███╔╝ ██║   ██║██║██╔████╔██║  "
+                "  ██║╚██╗██║██║ ██╔██╗ ╚██╗ ██╔╝██║██║╚██╔╝██║  "
+                "  ██║ ╚████║██║██╔╝ ██╗ ╚████╔╝ ██║██║ ╚═╝ ██║  "
+                "  ╚═╝  ╚═══╝╚═╝╚═╝  ╚═╝  ╚═══╝  ╚═╝╚═╝     ╚═╝  "
+              ];
+              opts = {
+                position = "center";
+                hl = "Type";
+              };
+            }
+            {
+              type = "padding";
+              val = 2;
+            }
+            {
+              type = "group";
+              val = [
+                {
+                  shortcut = "SPC sf";
+                  desc = "  Find file";
+                  command = ":Telescope find_files <CR>";
+                }
+                {
+                  shortcut = "SPC sr";
+                  desc = "  Recently used files";
+                  command = ":Telescope oldfiles <CR>";
+                }
+                {
+                  shortcut = "SPC sg";
+                  desc = "  Find text";
+                  command = ":Telescope live_grep <CR>";
```

Comment:
These default shortcuts are dependent on Telescope and will fail if Telescope is disabled.

Reply:
Yes but I don't feel like we should do anything here.
I would find it a bit over-killed to automatically install telescope for instance.

Reply:
I agree, we don't need to go as far as installing Telescope with Alpha. Though more minimal shortcuts are appropriate.